### ISSUE TITLE
Mounting shares as www-data breaks core_dev extension

### ISSUE THREAD

**Issue by kadamwhite** [Chassis/Chassis#691]
#663 breaks the ability to use the core_dev extension to run the WordPress core unit test suite within the virtual machine. Because the entire `wordpress-develop` directory is owned by www-data, `chmod` commands within the test suite fail when run via phpunit.

To replicate, follow the [use an existing WP checkout](https://github.com/Chassis/core_dev#use-an-existing-wordpress-checkout) instructions from the `core_dev` README, then provision your virtual machine and attempt to run 

```sh
vagrant ssh -c "cd /vagrant/wordpress-develop && phpunit --colors --filter Tests_WP_Customize_Manager"
```
to execute one specific test suite within the VM. You will see the `chown` command fail.

Tested in Vagrant 2.2.4 & 2.2.5 on an Ubuntu 19.04 host with VirtualBox 6.0.10.

**Comment by BronsonQuick** [comment]
<img width="967" alt="wordpress-core-dev__fish_VolumesSiteswordpress-core-dev__-fish__15841_2019-08-28_17-38-03" src="https://user-images.githubusercontent.com/1377956/63835351-a02e4e00-c9ba-11e9-8ce7-925e65f2402b.png">

I just tried this with the latest code in Chassis core and it worked for me. I'll destroy my box and do a fresh `vagrant up` and see if I can replicate it.

**Comment by BronsonQuick** [comment]
Damn. I had no luck replicating it on a fresh `vagrant up` either. I'm using Vagrant 2.2.5 and VirtualBox 6.0.10. I wonder if this is an Ubuntu specific issue? I literally only threw out my old laptop that had Ubuntu and Windows 10 yesterday so I can't try and replicate it 😢

**Comment by rmccue** [comment]
At a guess, could be because the underlying system has a `www-data` user/group, so it's trying to inherit those.

**Comment by ntwb** [comment]
> I'll destroy my box

I would too with that terminal font, what gives?

**Comment by BronsonQuick** [comment]
@ntwb Don't knock 'Input' 😆

**Comment by ntwb** [comment]
I'm not knocking _input_, used it for years, but something ain't right in that terminal window, the letter spacing is wide enough to drive _comic sans_ between  characters...

**Comment by rmccue** [comment]
Yeah, something is definitely wonky with your font rendering, this is what Input is _meant_ to look like:

<img width="546" alt="Screen Shot 2019-08-28 at 17 30 33" src="https://user-images.githubusercontent.com/21655/63874533-9bb56580-c9b9-11e9-80b7-1efb1e86f3c2.png">

Anyway, feel like we might be somewhat off-topic here 😜

**Comment by BronsonQuick** [comment]
Alrighty, so I just realised what the difference is. I didn’t have the `synced_folders:` on my install so my setup was as follows:

My folder structure is this:

Path | Description
---- | -------
`wordpress-develop/` | `wordpress-develop` git checkout
`wordpress-core-dev/` | Chassis root
 
<img width="1001" alt="wordpress-core-dev_VolumesSiteswordpress-core-dev_-_ config local yaml_2019-08-29_09-04-57" src="https://user-images.githubusercontent.com/1377956/63898385-26d73f80-ca3c-11e9-9154-fe5d284abf21.png">

My `paths` in `config.local.yaml` are this:
```
paths:
  # Use Chassis paths as normal,
  content: content
  base: .
  # Except use your checkout for the WordPress directory. To switch
  # between /src and /build, edit this line then run `vagrant provision`.
  wp: wordpress-develop/src
```
and my relevant lines from `mount` 
```
vagrant on /vagrant type vboxsf (rw,nodev,relatime,iocharset=utf8,uid=33,gid=33,dmode=0777,fmode=0777)
var_log_php on /var/log/php type vboxsf (rw,nodev,relatime,iocharset=utf8,uid=33,gid=33,dmode=0777,fmode=0777)
var_log_nginx on /var/log/nginx type vboxsf (rw,nodev,relatime,iocharset=utf8,uid=33,gid=33,dmode=0777,fmode=0777)
vagrant_extensions__global on /vagrant/extensions/_global type vboxsf (rw,nodev,relatime,iocharset=utf8,uid=33,gid=33,dmode=0777,fmode=0777)
tmp_vagrant-puppet_manifests-846018e2aa141a5eb79a64b4015fc5f3 on /tmp/vagrant-puppet/manifests-846018e2aa141a5eb79a64b4015fc5f3 type vboxsf (rw,nodev,relatime,iocharset=utf8,uid=0,gid=1000)
```

As you can see the `wordpress-develop` folder isn’t mounted so I’ve changed that and run `vagrant reload` so that the folders are mounted and I get the following:

```
vagrant on /vagrant type vboxsf (rw,nodev,relatime,iocharset=utf8,uid=33,gid=33,dmode=0777,fmode=0777)
vagrant_wordpress-develop on /vagrant/wordpress-develop type vboxsf (rw,nodev,relatime,iocharset=utf8,uid=33,gid=33,dmode=0777,fmode=0777)
vagrant_extensions__global on /vagrant/extensions/_global type vboxsf (rw,nodev,relatime,iocharset=utf8,uid=33,gid=33,dmode=0777,fmode=0777)
```

<img width="1440" alt="wordpress-core-dev__fish_VolumesSiteswordpress-core-dev__-fish__15841_2019-08-29_09-02-36 (1)" src="https://user-images.githubusercontent.com/1377956/63898407-3c4c6980-ca3c-11e9-8f6a-dcecb2b331f3.png">

So, given all of that new information I'm guessing the difference might be that you had an existing `wordpress-develop` checkout but mine was added automagically by https://github.com/Chassis/core_dev/blob/master/modules/core_dev/manifests/repository.pp#L52

Does that sounds correct? If that's the case can you let me know how you cloned your original `wordpress-develop` and from which repo? I'm not sure if you used Git or SVN or if you used the official WP git or the WP Github mirror. Taa muchly!

Oh yeah and here's some more output.
<img width="701" alt="wordpress-core-dev__vagrantcore___ssh__vagrant_ssh__15841_2019-08-29_09-10-14" src="https://user-images.githubusercontent.com/1377956/63898580-e5935f80-ca3c-11e9-877e-5bca3ffb2545.png">

**Comment by kadamwhite** [comment]
@BronsonQuick you’re correct, I had an existing core checkout; it’s a clone of the official WP git mirror (don’t have the link handy but it’s on make/core), cloned into a sibling directory of the chassis root. Having things set up this way is a strong advantage of the core dev extension as it was, since I can easily switch between SVN and Git checkouts; so I hope it will continue to be supported.

**Comment by BronsonQuick** [comment]
@kadamwhite Coolness. Thanks. I'm having a bit more of a dig into it now.

I had Dzikri give it a test on his Linux box and this is the error he got:
```
==> default: Error: Execution of 'git clone git://develop.git.wordpress.org/ /vagrant/wordpress-develop' returned 128: Cloning into '/vagrant/wordpress-develop'...
==> default: error: chmod on /vagrant/wordpress-develop/.git/config.lock failed: Operation not permitted
==> default: error: chmod on /vagrant/wordpress-develop/.git/config.lock failed: Operation not permitted
==> default: error: chmod on /vagrant/wordpress-develop/.git/config.lock failed: Operation not permitted
==> default: error: chmod on /vagrant/wordpress-develop/.git/config.lock failed: Operation not permitted
==> default: fatal: 'origin' does not appear to be a git repository
==> default: fatal: Could not read from remote repository.
```

**Comment by BronsonQuick** [comment]
@kadamwhite Coolness. So I can finally replicate it now:
```
vagrant ssh -c "cd /vagrant/wordpress-develop && phpunit --colors --filter Tests_WP_Customize_Manager"
Installing...
Running as single site... To run multisite, use -c tests/phpunit/multisite.xml
Not running ajax tests. To execute these, use --group ajax.
Not running ms-files tests. To execute these, use --group ms-files.
Not running external-http tests. To execute these, use --group external-http.
PHPUnit 7.5.15 by Sebastian Bergmann and contributors.

.............E................................................... 65 / 67 ( 97%)
..                                                                67 / 67 (100%)

Time: 7.98 seconds, Memory: 172.50 MB

There was 1 error:

1) Tests_WP_Customize_Manager::test_import_theme_starter_content
chmod(): Operation not permitted

/vagrant/wordpress-develop/build/wp-admin/includes/file.php:881
/vagrant/wordpress-develop/build/wp-admin/includes/file.php:970
/vagrant/wordpress-develop/build/wp-admin/includes/media.php:434
/vagrant/wordpress-develop/build/wp-includes/class-wp-customize-manager.php:1385
/vagrant/wordpress-develop/tests/phpunit/tests/customize/manager.php:653
```

I'll have a think about it overnight and see if I can come up with a solution tomorrow that doesn't break the logs.

**Comment by kadamwhite** [comment]
Thanks for this; finally got a chance to test and yes, it works a treat! :heart:
